fontir/src/ir/static_metadata.rs

```diff
@@ -225,6 +227,60 @@ pub struct Panose {
     pub x_height: u8,
 }
 
+/// A series of substitution rules to be applied to layout features
+/// at specific points in design space.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
```

Comment:
Despite being in IR I think it would be reasonable to link to https://fonttools.readthedocs.io/en/stable/designspaceLib/xml.html#rules-element as an example of a likely source.

As it stands I think this is quite designspace specific which makes me wonder if there is a way to generalize the IR representation? - not a blocking issue tho.

Reply:
The designspace types pretty closely reflect the structure of the FeatureVariations table itself, so I hope that this will generalize; we'll find out how true that is when I move on to glyphs bracket layers :)

Reply:
Designspace rules only model a particular subset of what FeatureVariations tables can do, specifically single glyph substitutions under specific variation regions. FeatureVariations allow to replace any feature, not only rvrn or rclt or other simple GSUB features for single substitutions. An entire feature record gets replaced under certain conditions (if the current instance falls within a specified region of the variation space), this can be any kind of feature, including GPOS features.

Reply:
> when I move on to glyphs bracket layers

those are essentially the same thing as Designspace rules, only that instead of there being distinct alternate glyphs that get substituted as is the case with DS rules, there are "alternate layers" (this is the current naming convention for what used to be called 'bracket' layers) associated to some other glyph master layers, and the compiler needs to create glyphs for each of them (GSUB substitutes glyphs)